pco/src/float_quant_utils.rs

```diff
@@ -54,48 +56,46 @@ pub(crate) fn split_latents<F: FloatLike>(page_nums: &[F], k: Bitlen) -> Vec<Vec
   vec![primary, secondary]
 }
 
-pub(crate) fn compute_bid_w_sample<F: FloatLike>(sample: &[F]) -> Bid<F::L> {
-  let k = estimate_best_k(sample);
+pub(crate) fn compute_bid<F: FloatLike>(sample: &[F]) -> Option<Bid<F>> {
+  let (k, freq) = estimate_best_k_and_freq(sample);
   // Nothing fancy, we simply estimate that quantizing by k bits results in saving k bits per
-  // number.  This is based on the assumption that FloatQuant will usually be used on datasets that
-  // are exactly quantized.
+  // number, whenever possible. This is based on the assumption that FloatQuant
+  // will usually be used on datasets that are exactly quantized.
+  let bits_saved_per_num = (k as f64) * freq;
   if (k as f64) > QUANT_REQUIRED_BITS_SAVED_PER_NUM {
-    Bid::Candidate {
+    Some(Bid {
       mode: Mode::FloatQuant(k),
-      bits_saved_per_num: k as f64,
-    }
+      bits_saved_per_num,
+      split_fn: Box::new(move |nums| split_latents(nums, k)),
```

Comment:
Note to self: ah, gotta look up `move` -- probably related to some combination of `std::move` and rvalue references in C++ but I bet there are differences

Reply:
I think the idea is that it takes ownership of any variables it needs for the closure, as opposed to capturing them by reference (in this case `k`).

Reply:
Ah cool.  I would have expected each argument to have its own such qualifier (as iirc they do in C++) rather than a single one for all of them.